test/rules/prefer-conditional-expression/check-else-if/test.ts.lint

```diff
@@ -51,4 +51,33 @@ if (true) {
     foo(bar).baz = 1;
 }
 
+// checks return statements
+if (true) {
+~~ [err2]
+    return 1;
+} else {
+    return 2;
+}
+
+if (true) {
+    return 1;
+} else if (true) {
+       ~~ [err2]
```

Comment:
IMO the error should be located only at the first `if`

Reply:
I added it there because only the last set of such statements, IMO, should be changed to a conditional expression. When you have more than two return statements in an if-elseif-block, it would quickly get messy, and I thought that it would be clearer to show it there.

```javascript
if (x) {
    return 1;
} else if (y) {
   return 2;
} else {
   return 3;
}

// would become
if (x) {
   return 1;
}
return y ? 2 : 3;

// instead of
return x ? 1 : (y ? 2 : 3);
```

Reply:
That's why we have the option `"check-else-if"`. If enabled, you need to write these nasty nested conditional expressions. If not enabled, the rule completely ignores such cases